src/index.tsx

```diff
@@ -21,11 +21,19 @@ const FileUtils = NativeModules.FileUtils
 /**
  * Gets the duration of the video in seconds.
  * @param uri The full on device uri for the media item.
- * @param mediaType - Either 'image' or 'video. If passing an image, 0 will always be returned.
  * @returns Duration of the video in number of seconds.
  */
-export function getDuration(uri: string): Promise<number> {
-  return FileUtils.getDuration(uri);
+export async function getDuration(
+  uri: string,
+  mediaType: 'video' | 'image'
+): Promise<number> {
+  try {
+    if (mediaType !== 'video') return 0;
+    return FileUtils.getDuration(uri);
+  } catch (error) {
+    console.warn(`Can not get duration for ${uri}`);
+    return 0;
+  }
```

Comment:
Let's not cache the error at this point, I think the library should throw the error (maybe after some processing), then the client application will catch the error and return it safely. As there could be times where we would want the error to bubble up the the client layer.

Reply:
Removed 🗑️